### PR TITLE
Unit tests for APIPublisherImpl

### PR DESCRIPTION
**Pull request by tharikaGitHub**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/SampleTestObjectCreator.java [hunk 1]
```diff
@@ -150,7 +151,7 @@ public static LifecycleState getMockLifecycleStateObject(String lifecycleId) {
         LifecycleState lifecycleState = new LifecycleState();
         lifecycleState.setLcName("API_LIFECYCLE");
         lifecycleState.setLifecycleId(lifecycleId);
-        lifecycleState.setState("PUBLISH");
+        lifecycleState.setState("PUBLISHED");
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/SampleTestObjectCreator.java]: Use APIStatus constants

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/SampleTestObjectCreator.java [hunk 2]
```diff
@@ -150,7 +151,7 @@ public static LifecycleState getMockLifecycleStateObject(String lifecycleId) {
         LifecycleState lifecycleState = new LifecycleState();
         lifecycleState.setLcName("API_LIFECYCLE");
```
Comment (chamilaadhi) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/SampleTestObjectCreator.java]: use APIMgtConstants.API_LIFECYCLE

**Comment by tharikaGitHub** [comment]
Thanks for the review. Will do changes.

**Comment by chamilaadhi** [comment]
There are some strings constants in SampleTestObjectCreator.java class (not coming from your PR) . Shall we move them to constants and use existing constants where we can

**Comment by tharikaGitHub** [comment]
Yes, sure. I will move them.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/SampleTestObjectCreator.java [hunk 3]
```diff
@@ -20,11 +20,13 @@
 
 package org.wso2.carbon.apimgt.core;
 
+
 import org.apache.commons.collections.map.HashedMap;
```
Comment (pubudu538) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/SampleTestObjectCreator.java]: Shall we remove the new line?

Comment (tharikaGitHub) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/SampleTestObjectCreator.java]: Will remove it.

**Comment by tharikaGitHub** [comment]
Thanks for the review. Will rebase the code.

**Pull request merged**